app/controllers/correspondence_controller.rb

```diff
@@ -40,18 +43,28 @@ def verify_feature_toggle
     end
   end
 
-  def correspondence_load
-    @correspondence ||= correspondence_by_uuid
-    vet = veteran_by_correspondence
-    @all_correspondence = Correspondence.where(veteran_id: vet.id).where.not(uuid: params[:correspondence_uuid])
+  def correspondence
+    return @correspondence if @correspondence.present?
+
+    if params[:id].present?
+      @correspondence = Correspondence.find(params[:id])
+    elsif params[:correspondence_uuid].present?
+      @correspondence = Correspondence.find_by(uuid: params[:correspondence_uuid])
+    end
+
+    @correspondence
   end
 
-  def correspondence_by_uuid
-    Correspondence.find_by(uuid: params[:correspondence_uuid])
+  def correspondence_load
+    Correspondence.where(veteran_id: vet.id).where.not(uuid: params[:correspondence_uuid])
```

Comment:
@Dev-KRedd - I don't think `vet` is defined? I think this needs to be:

```suggestion
    Correspondence.where(veteran_id: veteran_by_correspondence.id).where.not(uuid: params[:correspondence_uuid])
```